In [38]:
import pandas as pd
import numpy as np
import numbers

In [2]:
# load the integrated dataset
df = pd.read_csv('NZ_Data/integrated_data.csv')

#### Check that the data of each field matches the expected datatype.

The columns with the object datatype are the ones that have mixed datatypes, so I'm going to review all of the data in them to make sure that the types are correct.

In [6]:
obj_cols = df.select_dtypes(include='object').columns

In [8]:
obj_cols= list(obj_cols)

In [9]:
obj_cols

['LANDWATER_NAME',
 'SA22018_name',
 'SVI18_CDEM_region',
 'SVI18_TA2018',
 'SVI18_SA2_name',
 'SVI18_urbanrural',
 'SVI18_HHchild0-4y_quality',
 'SVI18_HHchild0-14y_quality',
 'SVI18_single65overHHquality',
 'SVI18_singleparentHH_quality',
 'SVI18_nocar_quality',
 'SVI18_onepersonHH_quality',
 'SVI18_nomobile_quality',
 'SVI18_nointernet_quality',
 'SVI18_crowdedhh_quality',
 'SVI18_crowdedhhpeople_quality',
 'SVI18_damp_quality',
 'SVI18_severedamp_quality',
 'SVI18_mouldy_quality',
 'SVI18_severemould_quality',
 'SVI18_nosafewater_quality',
 'SVI18_nofridge_quality',
 'SVI18_nopower_quality',
 'Year']

All of the above columns have string data, so their classification as object data makes sense.

#### Check that numerical data is within an expected range

All numerical data should be positive

In [39]:
for index, row in df.iterrows():
    for col in cols:
        if isinstance(row[col], numbers.Number) and row[col] < 0:
            print(col)
            print(index)
            print(row[col])

All percentages should be between 0 and 100 - PROBLEM

In [59]:
cols = df.columns
for index, row in df.iterrows():
    for col in cols:
        if 'percent' in col:
            if row[col] < 0 or row[col] > 100:
                print(col)
                print(index)
                print(row[col])
                print(row["SA12018_V1_00"])
    

SVI18_o60yEuropean_percent
23827
104.5
7023827
SVI18_o60yEuropean_percent
23828
104.5
7023828
SVI18_o60yEuropean_percent
23829
104.5
7023829


count = 69; denom = 66

In [60]:
df.at[df.index[23827],'SVI18_o60yEuropean_percent'] = 100.0
df.at[df.index[23828],'SVI18_o60yEuropean_percent'] = 100.0
df.at[df.index[23829],'SVI18_o60yEuropean_percent'] = 100.0

In [61]:
cols = df.columns
for index, row in df.iterrows():
    for col in cols:
        if 'percent' in col:
            if row[col] < 0 or row[col] > 100:
                print(col)
                print(index)
                print(row[col])
                print(row["SA12018_V1_00"])
    

In [43]:
for index, row in df.iterrows():
    for col in cols:
        if 'Prp' in col:
            if row[col] < 0 or row[col] > 100:
                print(col)
                print(index)
                print(row[col])

In [63]:
df = df.drop('SA12018_V1', axis = 1)


In [66]:
df.to_csv('NZ_Data/integrated_checked.csv')

#### Check that SA1 Codes are Unique

In [40]:
lst = list(df['SA12018_V1_00'])

In [41]:
st = set(lst)

In [42]:
len(lst) == len(st)

True

#### Next step is visual checks

In [58]:
df.at[df.index[29872],'C18_OccPrivDwelDamp_TotalWith']

nan

In [57]:
df.at[df.index[29872],'SA12018_V1_00']

7029872

In [52]:
C18_D = pd.read_csv('NZ_Data/NZ_2018_census_data/2018_Census_Dwelling_data_by_SA1.csv')

In [53]:
C18_D.at[C18_D.index[29871],'C18_OccPrivDwelDamp_TotalWith']

-999

In [55]:
C18_D.at[C18_D.index[29871],'SA12018_V1_00']

7029872

There were some differences between the map of dwelling's with dampness between the original data and the integrated data. However, this is because the -999 values in the integrated dataframe were converted to NaN which makes the integrated map MORE accurate.

The map of SVI data for single parent households seemed to be accurate.

From a cursory look, NZDep2018 map seems to be accurate as well.

I don't have anything to compare it too, but the 2022 industry data looks like I expected it to on a map